include/boost/url/string.hpp

```diff
@@ -20,33 +20,83 @@
 namespace boost {
 namespace urls {
 
-/** A read-only, reference counted string
+/** A read-only string
 
     Objects of this type represent read-only
     strings with ownership of the character
-    buffer. They are reference counted which
-    makes copies cheap. The type is derived
-    from @ref string_view which provides
-    compatibility with strings in terms of
-    comparisons and converisons. However,
-    care must be exercised; undefined
+    buffer.
+
+    They are reference counted to
+    makes copies more efficient.
```

Comment:
saying that copies are "more efficient" is worse than saying "cheap." More efficient could mean that copies are 10 times faster. Which is definitely more efficient, but is not "cheap" (consider copying a 10 megabyte string). "cheap" has a specific meaning in C++. For example, copying an iterator is supposed to be "cheap." Copying a string view is "cheap". etc.

Reply:
So should I say "cheap" or "constant time"?

Reply:
I like "cheap" because "constant time" sounds more precise, and I don't want to paint myself into a corner making complexity claims. Sometimes I do make complexity claims (for example in JSON) and in that case "constant time" is appropriate, but note that when you say "constant time" you are baking that into the public interface. If the algorithm changes to be no longer constant time, then that is technically breaking programs that depend on the complexity guarantee.

In short, "cheap" gives us some needed wiggle room and communicates the idea sufficiently, without locking us into making a guarantee.

Reply:
> "cheap" has a specific meaning in C++

> I like "cheap" because "constant time" sounds more precise

What's the specific meaning of "cheap" in C++ then? Less than O(log n)?